### PR TITLE
Fix: progress bar for read in order queries

### PR DESCRIPTION
**Pull request by devcrafter**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit ade0906910bf5aa7bb5433f3bbef411d44d0ab17 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65769/ade0906910bf5aa7bb5433f3bbef411d44d0ab17/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Processors/QueryPlan/ReadFromMergeTree.cpp [hunk 1]
```diff
@@ -565,7 +565,7 @@ Pipe ReadFromMergeTree::readInOrder(
     /// In this case we won't set approximate rows, because it will be accounted multiple times.
     /// Also do not count amount of read rows if we read in order of sorting key,
     /// because we don't know actual amount of read rows in case when limit is set.
-    bool set_rows_approx = !is_parallel_reading_from_replicas && !reader_settings.read_in_order;
+    bool set_rows_approx = !is_parallel_reading_from_replicas && !query_info.limit;
```
Comment (nickitat) [inline src/Processors/QueryPlan/ReadFromMergeTree.cpp:568]: pls adjust the comment

**Review by alexey-milovidov — APPROVED** [review]

**Comment by devcrafter** [comment]
AST Fuzz failure - https://github.com/ClickHouse/ClickHouse/issues/65867

**Comment by clickgapai** [comment]
Hi @devcrafter @alexey-milovidov — while reviewing this PR I found the following:

- **Bug (P2):** Per-part progress summing in `readInOrder` causes false `TOO_MANY_ROWS` for multi-part tables — https://github.com/ClickHouse/ClickHouse/issues/104898

Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**